### ISSUE TITLE
Using gatsby behind corporate Proxy

### ISSUE THREAD

**Issue by nicgutierrez** [gatsbyjs/gatsby#20148]
I'm currently working behind corporate proxy and every single request going out needs to go through the proxy. Now, although I have all my ENV variables setup correctly, it seems that `gatsby develop` is trying to retrieve data from sources bypassing the proxy.

I investigated this a bit more, and the issue seems to be with `node` itself not respecting proxy settings at all. I found some workarounds in the past for `nextjs` by using a custom server with the `global-tunnel-ng` package, however, I wouldn't know how to implement this in Gatsby or if this is feasible at all.

Going through the documentation I found out I could potentially use some sort of proxy settings like so: https://www.gatsbyjs.org/docs/api-proxy/ and maybe I could even pass my corporate proxy settings using `https-proxy-agent` as described here: https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/corporate-proxy.md , but I'm not sure if that will actually achieve what I'm after.

Basically, I would like to set my corporate proxy settings to make sure my requests to APIs in `gatsby-node.js` succeed instead of getting `"ECONNREFUSED"` errors.

**Comment by wardpeet** [comment]
Using nodejs and proxies doesn't work well together luckily there are packages to fix this. Like you mentioned `global-tunnel-ng` is one but it only works for node <10. If you use a newer version you should use `global-tunnel`. You can initiate this in gatsby-config.js

Put this at the top of your gatsby-config.js. Make sure to set `GLOBAL_AGENT_HTTP_PROXY` environment variable to point which proxy you want to use. Checkout https://github.com/gajus/global-agent for more info.
```js
const { bootstrap } = require("global-agent")

bootstrap()
```

I hope this helps

**Comment by nicgutierrez** [comment]
thanks @wardpeet - adding a global tunnel in the config file did the trick! thanks :)

**Comment by vicduraes** [comment]
@nicgutierrez this issue is a little old, but I found it searching about how gatsby works with corporate proxy. I tried this solution but got an SSL error.

` 
  Error: write EPROTO 140704729167040:error:1408F10B:SSL routines:ssl3_get_record:wrong versio
  n number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
 `
 
 Did you have this error at the time? 
 
 Thank you, this issue was the closest I got to solving my problem.
